File: tjhsst comp sci 2/Java/Units/Unit1/Lab05.java
//Name: Vivian Feng     Date: March 17,2020

import edu.fcps.karel2.Robot;
import edu.fcps.karel2.Display;

public class Lab05
{
   public static void runTheRace(Racer arg)
   {
      int[] beeperList = {7,5,3};
      
      /*for each pile:
         - go to pile and go back*/
      int spaces = 2;
      for (int pile=0;pile<3;pile++)
      {
         arg.shuttle(spaces,beeperList[pile]);
         spaces += 2;
      }
      arg.move();
   }
   
   public static void main(String[] args)
   {
      Display.openWorld("maps/shuttle.map");
      Display.setSize(10,10);
      Display.setSpeed(10);
      
      Racer bob = new Racer(1);
      Racer joe = new Racer(4);
      Racer rick = new Racer(7);
      
      runTheRace(bob);
      runTheRace(joe);
      runTheRace(rick);
   }
}

File: tjhsst comp sci 2/pyKarel Student Code/Lab05.py
#Name: Vivian and Karthik
#date: 9/6/2019

from pyKarel import *
from racers import Racer

wld = World("shuttle")

r1 = Racer(wld,1)
r2 = Racer(wld,4)
r3 = Racer(wld,7)

def runTheRace(arg):
   arg.jumpEast()
   arg.sprint(2)
   arg.pick(7)
   arg.sprint(2)
   arg.pick(5)
   arg.sprint(2)
   arg.pick(3)
   arg.turnAround()
   arg.sprint(5)
   arg.jumpWest()
   arg.turnRight()
   arg.move()
   arg.put(15)
   arg.turnAround()
   arg.move()
   
for racer in [r1,r2,r3]:
   runTheRace(racer)
   
   
wld.mainloop()